tensorflow/python/compiler/tensorrt/utils.py

```diff
@@ -257,3 +261,79 @@ def draw_graphdef_as_graphviz(graphdef, dot_output_filename):
   print("  - `sudo apt install -y graphviz`")
   print("  - `dot -Tpng <input_filename>.dot -o <output_filename>.png`")
   print("===================================================================\n")
+
+
+@functools.lru_cache
+def get_nvidia_gpu_desc_dict():
+
+  try:
+    all_devices = device_lib.list_local_devices()
+    cuda_devices = [d for d in all_devices if d.device_type == "GPU"]
+
+    local_devices = list()
+
+    for device in cuda_devices:
+      gpu_info = gpu_util.compute_capability_from_device_desc(device)
+      cc = gpu_info.compute_capability
+
+      if cc == (0, 0):
+        continue
+
+      local_devices.append({
+      	"name": device.name,
+      	"device": device.physical_device_desc,
+      	"compute_capability": ".".join([str(s) for s in cc])
+      })
+
+    return sorted(local_devices, key=lambda x: x["name"], reverse=False)
+
+  except errors_impl.NotFoundError as e:
+    if not all(x in str(e) for x in ["CUDA", "not find"]):
+      raise e
+
+    else:
+      logging.error(str(e))
+      return False
+
+
+@functools.lru_cache
+def is_platform_supported(precision, gpu_id=0):
+  """ This function determines if the current NVIDIA GPU platform supports the
```

Comment:
A docstring should start with a one-line summary, see [here](https://google.github.io/styleguide/pyguide.html#381-docstrings)